lib/config/hadeployconfigverify.go

```diff
@@ -0,0 +1,674 @@
+	}
+	if err := validateRequiredPathField(configInitials.SSHKeyFile, "ssh_key_file"); err != nil {
+		errorList.PushBack(err)
+	}
+
+	return getSingleErrorFromList(errorList)
+}
+
+func (c *HaDeployConfig) validateExistingInfraBackupConfig() error {
+	// validate existing infra backup config
+	backupConfig := c.Architecture.ExistingInfra.BackupConfig
+
+	if err := validateRequiredString(backupConfig, "backup_config", "object_storage", "file_system"); err != nil {
+		return err
+	}
+
+	if backupConfig == "file_system" {
+		if err := validateRequiredString(c.Architecture.ExistingInfra.BackupMount, "backup_mount"); err != nil {
+			return err
+		}
+	}
+
+	if backupConfig == "object_storage" {
+		if err := c.verifyObjectStorage(c.ObjectStorage.Config); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (c *HaDeployConfig) validateAwsBackupConfig() error {
+	// validate aws backup config
+	if err := checkForValidS3Bucket(c); err != nil {
+		return err
+	}
+	if c.Aws.Config.SetupManagedServices {
+		if err := validateRequiredString(c.Architecture.Aws.BackupConfig, "backup_config", "s3"); err != nil {
+			return err
+		}
+	} else {
+		if err := validateRequiredString(c.Architecture.Aws.BackupConfig, "backup_config", "s3", "efs"); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (c *HaDeployConfig) verifyObjectStorage(objectStorage *ConfigObjectStorage) error {
+	errorList := list.New()
+	if err := validateRequiredString(objectStorage.AccessKey, "access_key"); err != nil {
+		errorList.PushBack(err)
+	}
+	if err := validateRequiredString(objectStorage.SecretKey, "secret_key"); err != nil {
+		errorList.PushBack(err)
+	}
+	if err := validateRequiredString(objectStorage.BucketName, "bucket_name"); err != nil {
+		errorList.PushBack(err)
+	}
+	if err := validateS3Endpoint(objectStorage.Endpoint); err != nil {
+		errorList.PushBack(err)
+	}
+	if objectStorage.Region != "" {
+		if err := validateS3AWSRegion(objectStorage.Region); err != nil {
+			errorList.PushBack(err)
+		}
+	}
+	return getSingleErrorFromList(errorList)
+}
+
+func (c *HaDeployConfig) verifyAutomateSettings() error {
+	automateSettings := c.Automate.Config
+	errorList := list.New()
+
+	if err := validateFQDN(automateSettings.Fqdn, "automate fqdn"); err != nil {
+		errorList.PushBack(err)
+	}
+	if err := validateAutomateAdminPassword(automateSettings); err != nil {
+		errorList.PushBack(err)
+	}
+	if err := validateNumberField(automateSettings.InstanceCount, "automate instance_count", true); err != nil {
+		errorList.PushBack(err)
+	}
+	if err := validateRequiredString(automateSettings.ConfigFile, "config_file", "configs/automate.toml"); err != nil {
+		errorList.PushBack(err)
+	}
+	if err := validatePort(automateSettings.TeamsPort, "teams_port", false); err != nil {
+		errorList.PushBack(err)
+	}
+	if automateSettings.EnableCustomCerts {
+		if err := validateAutomateCerts(automateSettings); err != nil {
+			errorList.PushBack(err)
+		}
+		if c.IsExistingInfra() {
```

Comment:
we are going to support certs_by_ip for aws too in some time. just adding this comment as a note